features/guidance/notification.feature

```diff
@@ -0,0 +1,65 @@
+@routing  @guidance @mode-change
+Feature: Notification on turn onto mode change
+
+    Background:
+        Given the profile "car"
+        Given a grid size of 20 meters
+
+    Scenario: Turn onto a Ferry
+        Given the node map
+            |   |   |   |   |   | f |
+            | b | c |   |   | d | e |
+            |   |   |   |   |   |   |
+            | a |   |   |   |   | g |
+
+        And the ways
+            | nodes | highway | route | name  |
+            | ab    | primary |       | bingo |
+            | bc    | primary |       | dingo |
+            | cde   |         | ferry | ferry |
+            | eg    | primary |       | wingo |
+
+        When I route I should get
+            | waypoints | route                   | turns                                                   | modes                               |
+            | a,g       | bingo,ferry,wingo,wingo | depart,notification straight,notification right,arrive  | driving,ferry,driving,driving       |
+
+    Scenario: Turn onto a Ferry
+        Given the node map
+            |   |   |   |   |   |   | g |
+            |   | a |   |   |   |   | e |
+            |   |   | c |   |   | d |   |
+            |   |   |   |   |   |   | f |
+            |   | b |   |   |   |   |   |
+
+        And the ways
+            | nodes | highway | route | name  | oneway |
+            | ca    | primary |       | bingo | yes    |
+            | bc    | primary |       | dingo | yes    |
+            | cde   |         | ferry | ferry | no     |
+            | cdf   |         | ferry | ferry | no     |
+            | ge    | primary |       | wingo | no     |
+
+        When I route I should get
+            | waypoints | route                         | turns                                                               | modes                                   |
+            | g,a       | wingo,ferry,ferry,bingo,bingo | depart,notification right,continue right,notification right,arrive  | driving,ferry,ferry,driving,driving     |
+            | b,g       | dingo,ferry,ferry,wingo,wingo | depart,turn right,continue left,notification left,arrive            | driving,ferry,ferry,driving,driving     |
+
+    Scenario: Straight onto a Ferry
+        Given the node map
+            |   |   |   |   |   |   |   |   |
+            | a | b | c |   |   | d |   | i |
+            |   |   |   |   |   |   |   |   |
+            |   |   |   |   |   |   | f |   |
+
+        And the ways
+            | nodes | highway | route | name    | oneway |
+            | ab    | primary |       | yonkers | yes    |
+            | bc    | primary |       | dingo   | yes    |
+            | ca    | primary |       | bingo   | yes    |
```

Comment:
again, having three roads of different names here is going to pose problems at some point. Would you assume this to be a realistic setting?